Orleans/src/Microsoft.Orleans.Host.Abstractions/AzureStorageProvider.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Orleans.Host.Abstractions
+{
+    // this would go into the azure storage providers DLL
+    public static class AzureStorageProviderExtensions
```

Comment:
extensions should go in an {namespace}.Extensions namespace according to framework standards.  Users should have the option to opt in.